### PR TITLE
Validation with react-payer, return list of rejects at bottom of play…

### PR DESCRIPTION
**Pull request by g-a-v-i-n**
…list

- [x] validate youtube playlist - grab current track not playlist regex etc
- [x] give validation rejection message on reject track
- [X] use react-player validation
- [x] merge master and fix conflicts
- [x] move search stuff into main state
- [x] playlist title link and style

### CODE REVIEW

**Comment by g-a-v-i-n** [comment]
This is ready!

File: src/Main.js [hunk 1]
```diff
@@ -53,22 +54,39 @@ class Main extends Component {
...
+        const resortedPlaylists = {
+          ...playlistChannel,
+          contents: playlistChannel.contents.reverse()
+        }
         this.setState({
           playlistListLength: length,
-          playlists,
-          maxItemsInCurrentPage: this.getMaxItemsInCurrentPage(length, this.state.per),
+          playlistChannel: resortedPlaylists,
+          searchList: playlistChannel,
```
Comment (broskoski) [inline src/Main.js:69]: 😎

Comment (g-a-v-i-n) [reply src/Main.js:69]: hmm i'm not sure searchlist is actually used now that i see this

File: src/components/Header.js [hunk 2]
```diff
@@ -2,31 +2,50 @@ import React, { Component } from 'react'
...
       return ''
     }
     return ''
   }
 
+  handleToArena = () => {
+    const { currentOpenPlaylist, isCurrentPlaylistLoaded } = this.props
+    if (currentOpenPlaylist && isCurrentPlaylistLoaded && this.props.location.pathname !== '/') {
+      return (
+        <a target={'_blank'} href={`https://www.are.na/${currentOpenPlaylist.user.slug}/${currentOpenPlaylist.slug}`}>
```
Comment (broskoski) [inline src/components/Header.js:26]: Very nice

File: src/lib/helpers.js [hunk 3]
```diff
@@ -6,43 +8,79 @@ function makeHash() {
...
+function mm(url, message, item) {
+  return { url: url, message: message, item: item }
 }
 
-function classifyItem(item) {
-  const isAttachment = item.class === 'Attachment'
-  const isMedia = item.class === "Media"
+// our default messages
+const message = {
+  missing: 'Missing URL 😮',
```
Comment (broskoski) [inline src/lib/helpers.js:29]: Yes yes yes

File: src/Main.js [hunk 4]
```diff
@@ -53,22 +54,39 @@ class Main extends Component {
...
+  togglePlaylistOrder = (resortedPlaylists) => {
+    this.setState({ resortedPlaylists: resortedPlaylists.reverse() })
+  }
+
+  applySearch = (predicate) => {
+    const updatedList = this.state.playlistChannel.contents.filter(item => {
+      const text = decode(`${item.user.full_name} / ${item.title}`)
+      return text.toLowerCase().search(predicate) !== -1
+    })
+    this.setState({ searchList: { ...this.state.playlistChannel, contents: updatedList } })
```
Comment (broskoski) [inline src/Main.js:83]: Looks like it just gets replaced here, makes semi-sense to me (set the searchList to the entire channel contents to start)

**Comment by broskoski** [comment]
Looks fantastic, thank you a million!

File: src/Main.js (hunk 4 above)
Comment (g-a-v-i-n) [reply src/Main.js:83]: ah gotcha. funny to be reminded why you did something

**Pull request merged**